### ISSUE TITLE
Create non-production commitments

### ISSUE THREAD

**Issue by fosterlynn** [Carbon-Farm-Network/app-carbon-farm-network#127]
Lynn: I'm putting 1 week on this, pending specing it out.

Add the ability to create commitments for non-production exchanges, as well as other inputs/payments, such as yarn payments, fees, accountant, etc.  So this would include existing and new agreements, with primary and reciprocal commitments.

[EDIT: Removed cash flow feature to separate issue #154 ]

We need to decide how to include sample yarn production, since that is not currently part of the stages of production planning.  It also includes exchanges.

[EDIT: Removed events, due to separate milestone.  If easier to just leave it in due to copy/paste or modularization, please feel free to leave it in.]

**Comment by fosterlynn** [comment]
This could look something like this.  The * data is events, the other data is commitments.
```
Exchanges   (Add)   (select year?)  (filter finished/all?) (filter cash/production/all?)
Primary Planned/Actual               Reciprocal Planned/Actual
date-action-qty-resspec-from-to  date-action-qty-resspec-from-to
  * date-qty                      * date-qty
  * date-qty                      * date-qty
```
Edit capability on commitments.  Correct capability on events?

**Comment by fosterlynn** [comment]
Let's start this one so we can see what commitments are actually in the data store, will make it easier to test the planning page.  Also it will show if the Agreement structure is being saved properly.

So, let's create just a report of all Agreements, including both production and non-production commitments, with the primary Commitment and the reciprocal Commitment.  (I think there is just one of each as of now for CFN.  If just as easy, make it support multiples of either.)  Don't include any commitments that aren't part of an agreement.

So it would look something like this, the first two lines being headings:

Exchanges 
Primary Planned/Actual                       Reciprocal Planned/Actual
date-action-qty-resspec-from-to           date-action-qty-resspec-from-to

That last is to say create a phrase from the data.  Like:
```
11/12/2024 deliver-service 407 pounds Scouring Service Alpaca from Bainton's Old Mill to Carbon Farm Network       11/20/2024 transfer 814.00 USD from Carbon Farm Network to Bainton's Old Mill
```
Or feel free to re-do the format to make it fit better or more readable or whatever improvements are needed.

Note:  I'm adding the next label to this, but it just means for what is in this comment, not the whole report.  It can still get added to the top menu

**Comment by LeosPrograms** [comment]
Will this be it's own CRUD page with a link from the navbar?

**Comment by fosterlynn** [comment]
>Will this be it's own CRUD page with a link from the navbar?

Yes I think so, at least in the longer run.  However, for now, you can feel free to do whichever is easiest.  Name: Exchanges.

**Comment by fosterlynn** [comment]
First iteration (just showing the exchange commitments) looks good, taking off the labels and changing milestone.

**Comment by LeosPrograms** [comment]
Is what is remaining the ability to add an agreement from a modal?

**Comment by fosterlynn** [comment]
>Is what is remaining the ability to add an agreement from a modal?

Yes, at least that is what is remaining for this planning milestone.  

Then the question is whether to include Agreement, we don't now on the report.  For now, I think I'd like to do it in the simplest way possible.  So the choices (that I can think of, you may have other ideas) are:
* Keep it without the Agreement being shown or explicitly added, but add it under the covers, like we do now basically.  Still show Commitments that are connected by an Agreement as you do now.  Then the model is what?  2 Commitments on one modal?  Or the ability to keep adding Commitments within a modal?
* Do it like recipe exchanges, include the Agreement (for now, with name and note).  For this option, we wouldn't necessarily have to include the Agreement on the list itself, the list could remain as it is, it's good to have that level of detail on the Commitments.  In that case, the Edit/Delete would apply to the whole Agreement with its Commitments.

If they are somewhat equally easy, then I think the second option would be best, sets it up for the future better.

**Comment by fosterlynn** [comment]
By the way, this is the newer VF structure for Agreements.  Just in case it makes everything easier about Agreements.  If not, we can of course wait.  We'd probably need to discuss how to implement the inverse relationships in holochain.

![Image](https://github.com/user-attachments/assets/127830cf-0f71-4b8b-b581-2b140941785f)

**Comment by fosterlynn** [comment]
lookin' good
